### PR TITLE
In-memory engine: add pre-load extra msg for pre-load when transfer leader

### PR DESCRIPTION
**Pull request by SpadeA-Tang**

### CODE REVIEW

**Comment by SpadeA-Tang** [comment]
@overvenus PTAL

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -255,6 +255,7 @@ enum ExtraMessageType {
     // Messages for the snapshot gen precheck process.
     MsgSnapGenPrecheckRequest = 15;
     MsgSnapGenPrecheckResponse = 16;
+    MsgPreLoadRegionRequest = 17;
```
Comment (overvenus) [inline proto/raft_serverpb.proto:258]: Please add some comments.

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Comment by SpadeA-Tang** [comment]
PTAL @glorv

**Review by glorv — APPROVED** [review]

**Pull request merged**